### PR TITLE
Find a better way to determine last status change

### LINKED ISSUE #1636: New TO server field: LastStatusChange
As a user, I want to know when the last status change of a server was so that I can track how long a server has been in a given status such as OFFLINE or ADMIN_DOWN.  The only current way to maybe do it is to parse the entire changelog, but that doesn't work with API changes where the individual changes aren't posted to the changelog.

### PR DESCRIPTION
**Pull request by srijeet0406**
## What does this PR (Pull Request) do?

- [x] This PR fixes #1636  

## Which Traffic Control components are affected by this PR?

- CDN in a Box
- Traffic Ops
- Traffic Portal
- CI tests

## What is the best way to verify this PR?

Run local TO and TP and create a server. Notice that the `status_last_updated` field gets set to the current time in the DB.
Now, go to TP, and select the `Status Last Updated` as one of the display columns for the servers table. Notice that it shows the correct time elapsed since you created the server.
Now, update this server changing any field, EXCEPT the `status` of the server. Notice that the `status_last_updated` value DOES NOT change.
Now, update the server by changing the server's status (you will have to do this from the backend call, since TP doesn't let you do it from the front end). 
Notice that the `status_last_updated` time gets changed to the current time this time around.

Also, make sure unit tests and API tests pass.

## If this is a bug fix, what versions of Traffic Control are affected?

- master

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR
- [x] This PR does not include documentation
- [x] This PR includes an update to CHANGELOG.md
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: lib/go-tc/servers.go [hunk 1]
```diff
@@ -342,45 +343,46 @@ type ServerV1 struct {
...
+	RouterPortName    *string              `json:"routerPortName" db:"router_port_name"`
+	Status            *string              `json:"status" db:"status"`
+	StatusID          *int                 `json:"statusId" db:"status_id"`
+	TCPPort           *int                 `json:"tcpPort" db:"tcp_port"`
+	Type              string               `json:"type" db:"server_type"`
+	TypeID            *int                 `json:"typeId" db:"server_type_id"`
+	UpdPending        *bool                `json:"updPending" db:"upd_pending"`
+	XMPPID            *string              `json:"xmppId" db:"xmpp_id"`
+	XMPPPasswd        *string              `json:"xmppPasswd" db:"xmpp_passwd"`
+	StatusLastUpdated time.Time            `json:"statusLastUpdated" db:"status_last_updated"`
```
Comment (ocket8888) [inline lib/go-tc/servers.go:385]: This, unfortunately, cannot be done. It changes the structures returned by all client methods. Those need to stay frozen at whatever they were prior to API v3, and this new field must be added only to the APIv3 Server object.

File: traffic_ops/traffic_ops_golang/server/put_status.go [hunk 2]
```diff
@@ -156,10 +157,12 @@ func updateServerStatusAndOfflineReason(serverID, statusID int, offlineReason *s
 	q := `
 UPDATE server
 SET    status = $1,
-       offline_reason = $2
-WHERE  id = $3
+       offline_reason = $2,
+       status_last_updated = $3
```
Comment (ocket8888) [inline traffic_ops/traffic_ops_golang/server/put_status.go:161]: You can actually just set this to `NOW()` right in the query, no need to interpolate.

**Comment by srijeet0406** [comment]
It actually does account for the changes made via `PUT /servers/{{ID}}`. But you're right about the user being able to change the `last status change time`. I'll fix that.

**Comment by srijeet0406** [comment]
Actually, scratch that. You cannot change the `status_last_updated` time by doing a `PUT` to `/servers/{{ID}}` because of this part of the code:

`if newServer.StatusID != nil && *newServer.StatusID != originalStatusID {
			newServer.StatusLastUpdated = time.Now()
		} else {
			newServer.StatusLastUpdated = origSer[0].StatusLastUpdated
		}`

**Comment by srijeet0406** [comment]
Closed in favor of https://github.com/apache/trafficcontrol/pull/4989

**Pull request closed without merging**